device_doctor/lib/src/android_device.dart

```diff
@@ -249,6 +256,51 @@ class AndroidDeviceDiscovery implements DeviceDiscovery {
     return healthCheckResult;
   }
 
+  /// The health check for battery level.
+  Future<HealthCheckResult> batteryLevelCheck({ProcessManager processManager}) async {
+    HealthCheckResult healthCheckResult;
+    try {
+      // The battery level returns two rows. For example:
+      //   level: 100
+      //   mod level: -1
+      final String levelResults = await eval('adb', <String>['shell', 'dumpsys', 'battery', '|', 'grep', 'level'],
+          processManager: processManager);
+      final List<String> levels = levelResults.trim().split('\n').toList();
+      final int level = int.parse(levels.where((element) => element.startsWith('level:')).single.split(':')[1].trim());
+      if (level < _kBatteryMinLevel) {
+        healthCheckResult =
+            HealthCheckResult.failure(kBatteryLevelCheckKey, 'Battery level ($level) is below $_kBatteryMinLevel');
+      } else {
+        healthCheckResult = HealthCheckResult.success(kBatteryLevelCheckKey);
+      }
+    } on BuildFailedError catch (error) {
+      healthCheckResult = HealthCheckResult.failure(kScreenSaverCheckKey, error.toString());
+    }
+    return healthCheckResult;
+  }
+
+  /// The health check for battery temperature.
+  Future<HealthCheckResult> batteryTemperatureCheck({ProcessManager processManager}) async {
```

Comment:
is this phone temperature rather than battery temperature?

Reply:
This is battery temperature, consistent with what is used in [chromium bot configs](https://chrome-internal.git.corp.google.com/infradata/config/+/refs/heads/main/configs/chromium-swarm/scripts/android.py#137).